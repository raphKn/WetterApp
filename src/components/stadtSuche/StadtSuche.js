import React from 'react';
import PropTypes from 'prop-types';
import { StyleSheet } from 'react-native';
import CustomSection from '../common/CustomSection';
import LabeledTextInput from '../input/LabeledTextInput';

const styles = StyleSheet.create({
    container: { flexDirection: 'row', justifyContent: 'space-between', alignItems: 'center' },
    text: {
        flex: 1,
    },
    button: {
        margin: 0,
    },
});

const StadtSuche = props => (<CustomSection style={styles.container} name="Suche" backgroundColor={'#ff6600'}>
    <LabeledTextInput style={styles.text} label="Stadt" onSubmit={props.onSubmit} defaultText={props.defaultText} returnKeyType="search" />
</CustomSection>);

StadtSuche.propTypes = {
    onSubmit: PropTypes.func.isRequired,
    defaultText: PropTypes.string,
};
StadtSuche.defaultProps = {
    defaultText: '',
};

export default StadtSuche;
