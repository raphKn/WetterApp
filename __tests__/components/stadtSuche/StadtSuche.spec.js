import React from 'react';
import renderer from 'react-test-renderer';
import StadtSuche from '../../../src/components/stadtSuche/StadtSuche';

const doNothing = () => {
};

describe('StadtSuche', () => {
    it('renders correctly in a snapshot', () => {
        const wrapper = renderer.create(
            <StadtSuche onSubmit={doNothing} defaultText="DefaultStadt" />);
        expect(wrapper).toMatchSnapshot();
    });
});
