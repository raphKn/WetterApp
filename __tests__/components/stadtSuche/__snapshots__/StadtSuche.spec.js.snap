// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StadtSuche renders correctly in a snapshot 1`] = `
<View
  style={
    Array [
      Object {
        "paddingTop": 12,
      },
      Object {
        "backgroundColor": "#ff6600",
      },
    ]
  }
>
  <View>
    <Text
      accessible={true}
      allowFontScaling={false}
      disabled={false}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
            "color": "#212121",
          },
          Object {
            "fontSize": 18,
            "paddingHorizontal": 18,
          },
        ]
      }
    >
      Suche
    </Text>
  </View>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "#E0E0E0",
          "paddingHorizontal": 18,
          "paddingVertical": 6,
        },
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "justifyContent": "space-between",
        },
      ]
    }
  >
    <View
      style={
        Object {
          "flex": 1,
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={false}
        disabled={false}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "backgroundColor": "transparent",
              "color": "#212121",
            },
            Object {
              "marginTop": 16,
            },
          ]
        }
      >
        Stadt
      </Text>
      <View
        style={
          Object {
            "borderBottomColor": "#BDBDBD",
            "borderBottomWidth": 1,
            "marginBottom": 8,
            "marginTop": 8,
            "paddingBottom": 8,
          }
        }
      >
        <TextInput
          onChangeText={[Function]}
          onSubmitEditing={[Function]}
          returnKeyType="search"
          style={
            Object {
              "color": "#212121",
            }
          }
          underlineColorAndroid="transparent"
        />
      </View>
    </View>
  </View>
</View>
`;
